src/vcf/record.cpp

```diff
@@ -459,62 +519,73 @@ namespace ebi
             number_matches = (values.size() == static_cast<size_t>(expected));
         } else if (expected == 0) {
             // There will be one empty value that needs to be specifically checked
-            number_matches = values.size() == 0 ||
-                    (values.size() == 1 && (values[0] == "0" || values[0] == "1"));
+            number_matches = values.size() == 0 || values.size() == 1;
         } else {
             // if number=".", then `expected` was set to -1, and it should always match, letting `number_matches` as true
         }
 
         if (!number_matches) {
-            raise(std::make_shared<Error>(line, field + " does not match the meta specification Number=" + number +
-                    ", expected " + std::to_string(expected) + " values"));
+            raise(std::make_shared<Error>(line, " specification Number=" + number + ". Contains " + std::to_string(values.size()) + " values, expected " + std::to_string(expected)));
         }
     }
 
-    void Record::check_field_type(std::string const & field,
-                                  std::vector<std::string> const & values,
+    void Record::check_value_type(std::string const & type, std::string const & value, std::string & message) const {
```

Comment:
I still find the format of these error messages a bit inconsistent. This is the output for a synthetic VCF file:

```
Line 7: INFO INT=1.5 does not match the meta specification Type=Integer. 1.5 (float) provided instead of Integer
Line 8: INFO VALIDATED=2 does not match the meta specification Type=Flag. A flag value must be 0 and 1
Line 9: INFO FLOAT=lalala does not match the meta specification Type=Float.
```

For the first error, the second sentence could be `An integer must not contain decimal digits`. 
For the second error, a flag value must be "0, 1 or none".
For all errors, the last sentence should end with a full stop.